app/models/spotlight/resource.rb

```diff
@@ -96,6 +98,18 @@ def commit
         Rails.logger.warn "Unable to commit to solr: #{e}"
       end
 
+      # Called when the record is destroyed.
+      def remove_from_index
+        return unless write?
+
+        blacklight_solr.delete_by_query(
+          "#{Spotlight::Engine.config.resource_global_id_field}:#{to_global_id}"
```

Comment:
I think there's probably another necessary component to this to deal with solr documents that are generated into the index from multiple resources. To start with, we may need to do additional bookkeeping to track resources -> solr ids in the database.

Reply:
So, we need to check that resource_global_id_field is only a single element and that single element is this resource?

Reply:
And do the work to make sure `resource_global_id_field` is multi-valued when necessary.